### PR TITLE
[infra-proxy-service] Remove dup GetVersion API

### PR DESCRIPTION
**Pull request by vsingh-msys**
### :nut_and_bolt: Description: What code changed, and why?

- `GetVersion` API had been added into the infra-proxy service but seems the version can be retrieved using Debug API. It's default added to every service. PR to remove dup `GetVersion` API.

### :chains: Related Resources
Closes: #4431 

### :+1: Definition of Done
- Remove duplicate GetVersion API from infra-proxy-service.

### :athletic_shoe: How to Build and Test the Change
- `rebuild components/infra-proxy-service`
- `rebuild components/automate-gateway`
- Removed
```
chef-automate dev grpcurl infra-proxy-service chef.automate.domain.infra_proxy.service.InfraProxyService.GetVersion

```

- Version can be fetched using:

```
chef-automate dev grpcurl infra-proxy-service chef.automate.api.debug.Debug.GetVersion
{
  "version": "20201105143246",
  "gitRef": "a9b69ba302db8ad2e56c009574dd82f84ff9f0c4",
  "goVersion": "go1.14"
}
```

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
Signed-off-by: Vivek Singh <vivek.singh@msystechnologies.com>

### CODE REVIEW

**Review by stevendanna — COMMENTED** [review]
I would be a big +1 on this cleanup if this API weren't part of the external API.  This is _technically_ a breaking API change.  My guess is very few users would notice though.  Is the infra-proxy-service API officially released yet?  If not, then we can make this change. If so, then we can't unfortunately.

**Comment by vsingh-msys** [comment]
Nope, `infra-proxy-service` not yet released officially. so we can safely review it. Thanks!

**Review by stevendanna — APPROVED** [review]

File: buf.yaml [hunk 1]
```diff
@@ -398,3 +398,4 @@ breaking:
     # it's the best/least bad option, you can add it here and remove it in a
     # later PR:
     # - external/oops/oops.proto
+    - external/infra_proxy/infra_proxy.proto
```
Comment (msorens) [inline buf.yaml:401]: Consider adding a "TODO" comment here saying this needs to be removed in a follow-up PR.

**Review by msorens — APPROVED** [review]
Just a couple comments.

File: components/infra-proxy-service/server/server.go [hunk 2]
```diff
@@ -16,6 +16,16 @@ import (
 	"github.com/chef/automate/lib/tracing"
 )
 
+// Server is an infra-proxy server
+type Server struct {
+	service *service.Service
+}
+
+// NewServer returns an infra-proxy server
+func NewServer(service *service.Service) *Server {
```
[the remark this answers is not in the document]
Comment (vsingh-msys) [reply components/infra-proxy-service/server/server.go:25]: It's used at https://github.com/chef/automate/blob/ec5a497d9022a57313eb1d1dfb6166fff4975e89/components/infra-proxy-service/server/server.go#L40

[the remark this answers is not in the document]
Comment (vsingh-msys) [reply components/infra-proxy-service/server/server.go:25]: As this is moved from `root.go` file.

**Pull request merged**